### PR TITLE
chore!: convert to composite action

### PR DESCRIPTION
**Pull request by mrcjkb**
This is an alternate approach to #37.

### CODE REVIEW

**Comment by mrcjkb** [comment]
@teto I think we should also get rid of the `build_type` input and remove the `gnumake` dependency, What do you think?

**Comment by mrcjkb** [comment]
[This approach works](https://github.com/mrcjkb/peek.nvim/actions/runs/4461139762/jobs/7834756786), too.

File: action.yml [hunk 1]
```diff
@@ -65,17 +60,22 @@ inputs:
...
-    - ${{ inputs.template }}
-    - ${{ inputs.upload }}
-    - ${{ inputs.license }}
+  using: "composite"
+  steps:
+    - uses: cachix/install-nix-action@v20
+    - run: nix profile install "${{ github.action_path }}#luarocks-tag-release-action"
+      shell: bash
+    - run: luarocks-tag-release-action
+      env:
```
Comment (teto) [inline action.yml:69]: why have you moved them to the env ?

**Comment by teto** [comment]
I am not a github action expert but I much prefer this approach !

File: action.yml (hunk 1 above)
Comment (mrcjkb) [reply action.yml:69]: It's a workaround for https://github.com/actions/runner/issues/665

**Review by teto — APPROVED** [review]

**Comment by mrcjkb** [comment]
> I am not a github action expert but I much prefer this approach !

I do too. When I first created this, I wanted to avoid making it a composite action because I thought users would have to manually add the install-nix-action to their workflow.

**Comment by teto** [comment]
> @teto I think we should also get rid of the build_type input and remove the gnumake dependency, What do you think?

I dont know about the build_type, but I would keep gnumake as a dependency since it is an optional dependency of luarocks and I dont think ubuntu or fedora provides it out of the box, it makes the experience smoother.

One could say cmake is also an optional dependency of luarocks but in practice, I've seen much less rockspecs relying on cmake and it's even kinda discouraged by lua authors which is why I had not added it in the first place.

Also I haven't compared perfs of this new approach but if it's more practical, I dont mind a performance hit.

**Comment by mrcjkb** [comment]
I'm going to run some tests without `gnumake`.

**Comment by mrcjkb** [comment]
It works fine without gnumake on ubuntu: https://github.com/mrcjkb/peek.nvim/actions/runs/4461231493/jobs/7834909583

**Comment by teto** [comment]
cool !

**Comment by mrcjkb** [comment]
It seems to have shaved about 30 s off (by comparison of the peek.nvim runs and this repo's "Test luarocks install" run.
I think docker workflows are the slowest...

**Comment by teto** [comment]
> It seems to have shaved about 30 s

damn that's great xD

**Pull request merged**